carbon-design-system/carbon-addons-iot-react
# feat(tablecolumn): add optional tooltip to table column

Closes #https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/2221

**Summary**

- Adds new tooltip prop to each table column and it will render this as a description of the tooltip

**Change List (commits, features, bugs, etc)**

- feat(TableCellRenderer): preserve the existing TableCell truncation UX inside the Data section, but if the tooltip property is passed, then show a TooltipDefinition (currently only supported for column headers)
- feat(tableCellRenderer.scss): make the header class match for the tooltip
- feat(TableHead/TableHeader): need to allow overflow if a tooltip is being used
- feat(TablePropTypes): expose the tooltip prop on the column

**Acceptance Test (how to verify the PR)**

- Check the new story that has a column tooltip

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify the existing overflow tooltips work for normal table cells
